circuits/cpp/src/aztec3/circuits/abis/call_context.hpp

```diff
@@ -27,11 +30,17 @@ template <typename NCT> struct CallContext {
     boolean is_static_call = false;
     boolean is_contract_deployment = false;
 
+    // for serialization, update with new fields
+    MSGPACK_FIELDS(msg_sender,
+                   storage_contract_address,
+                   portal_contract_address,
+                   is_delegate_call,
+                   is_static_call,
+                   is_contract_deployment);
     boolean operator==(CallContext<NCT> const& other) const
     {
-        return msg_sender == other.msg_sender && storage_contract_address == other.storage_contract_address &&
-               portal_contract_address == other.portal_contract_address && is_delegate_call == other.is_delegate_call &&
-               is_static_call == other.is_static_call && is_contract_deployment == other.is_contract_deployment;
+        // we can't use =default with a custom boolean, but we can use a msgpack-derived utility
+        return utils::msgpack_derived_equals<boolean>(*this, other);
     };
```

Comment:
Does it make sense to turn this into a macro altogether? Or even bundle it within `MSGPACK_FIELDS`? Same for the `ostream <<` mathod.

Reply:
For now I rather not surprise people too much, but eventually possibly - but it'd be kinda fiddly as we have a custom bool type to pass here